src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -221,41 +222,83 @@ private void LogRequestFinished(HostingApplication.Context context, long startTi
         }
     }
 
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026",
+        Justification = "The values being passed into Write have the commonly used properties being preserved with DynamicDependency.")]
```

Comment:
Here's how the runtime annotates these calls: https://github.com/dotnet/runtime/blob/1b3ef865fd71b898848a1a4b53bf062db622efa9/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs#L173-L174. The idea is that since the sink references everything they care about, this is trimmer safe.

Reply:
Thanks for the pointer. It looks like the only difference is `DynamicallyAccessedMemberTypes.PublicProperties` can be removed here. Is that right?

Reply:
👍🏽